lib/ui/gallery_app_bar_widget.dart

```diff
@@ -214,36 +214,34 @@ class _GalleryAppBarWidgetState extends State<GalleryAppBarWidget> {
     var collection = widget.collection;
     final dialog = createProgressDialog(context, "please wait...");
     await dialog.show();
-    if (collection == null) {
-      if (widget.type == GalleryAppBarType.local_folder) {
-        collection =
-            CollectionsService.instance.getCollectionForPath(widget.path);
-        if (collection == null) {
-          try {
-            collection = await CollectionsService.instance
-                .getOrCreateForPath(widget.path);
-          } catch (e, s) {
-            _logger.severe(e, s);
-            await dialog.hide();
-            showGenericErrorDialog(context);
-          }
+    try {
+      if (collection == null) {
+        if (widget.type == GalleryAppBarType.local_folder) {
+          collection =
+              CollectionsService.instance.getCollectionForPath(widget.path);
+          collection ??=
```

Comment:
Why the duplicate call?

Reply:
I am not really sure, didn't touch the previous logic. After seeing the logic, I can see that we don't need line 220/21. Will remove it?